daemon/cmd/state.go

```diff
@@ -338,6 +351,13 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
 	}
 	endpointCleanupCompleted.Wait()
 
+	go func() {
+		for _, ep := range state.restored {
```

Comment:
It seems that we have 2 gors reading `restored` and another writing. 
Could this yield a race on the `restored` state?

Reply:
`state.restored` is intended to be created in a single goroutine (the main goroutine running Daemon bootstrap) before it is handed off to any other goroutines, after which point it is no mutated. At the added block above it is not modified, indeed right below there are other goroutines reading from if as well (`len(state.restored)`).

There would be a race if `state.restored` is mutated after it has been passed to another goroutine, but that should not be the case.

Reply:
Thank you for the details Jarno! I wasn't sure about the synchronization here. I was more concerned about the readers potentially reading stale state.